lib/src/rules/avoid_relative_lib_imports.dart

```diff
@@ -0,0 +1,76 @@
+// Copyright (c)  2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:linter/src/analyzer.dart';
+import 'package:linter/src/util/dart_type_utilities.dart';
+
+const _desc = r'Avoid relative imports for files in `lib/`.';
+
+const _details = r'''*DO* avoid relative imports for files in `lib/`.
+
+When mixing relative and absolute imports it's possible to create confusion 
+where the same member gets imported in two different ways.  An easy way to avoid 
+that is to ensure you have no relative imports that include `lib/` in their 
+paths.
+
+**GOOD:**
+
+```
+import 'package:foo/bar.dart';
+
+import 'baz.dart';
+
+...
+```
+
+**BAD:**
+
+```
+import 'package:foo/bar.dart';
+
+import '../lib/baz.dart';
+
+...
+```
+
+''';
+
+class AvoidRelativeLibImports extends LintRule {
+  AvoidRelativeLibImports() : super(
+          name: 'avoid_relative_lib_imports',
+            description: _desc,
```

Comment:
Is this formatted?

Reply:
Whoops!  No.  Fixed now (but there should really be a presubmit!)